### PR TITLE
Fix scripts for docker build on mac

### PR DESCRIPTION
**Pull request by ccaraman**
Mac users require running the docker command with root permission.

### CODE REVIEW

**Comment by ccaraman** [comment]
@lyft/network-team

File: ci/README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ generate the binary in `~/build/envoy/source/exe/envoy-fastbuild` you can run:
 
 
 ```bash
-ENVOY_DOCKER_BUILD_DIR=~/build ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev.server_only'
+ENVOY_DOCKER_BUILD_DIR=~/build ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.debug.server_only'
```
Comment (mattklein123) [inline ci/README.md]: This doesn't actually match the output directory above AFAIK. I would just switch back to dev and remove server_only

File: docs/install/sandboxes.rst [hunk 2]
```diff
@@ -297,10 +297,13 @@ This image has all software needed to build envoy. From your envoy directory::
 
   $ pwd
   src/envoy
-  $ docker run -t -i -v <SOURCE_DIR>:/source lyft/envoy-build:latest /bin/bash -c "cd /source && ci/do_ci.sh normal"
+  $ ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev'
 
 That command will take some time to run because it is compiling an envoy binary.
 
+For more information on building, please refer to
+`this documentation <https://github.com/lyft/envoy/tree/master/ci#building-and-running-tests-as-a-developer>`_.
```
Comment (mattklein123) [inline docs/install/sandboxes.rst]: nit: you can use ```:repo:``` here (search other docs)

File: ci/README.md (hunk 1 above)
Comment (htuch) [inline ci/README.md]: +1, thanks!

File: ci/README.md [hunk 3]
```diff
@@ -37,7 +37,7 @@ generate the binary in `~/build/envoy/source/exe/envoy-fastbuild` you can run:
 
 
 ```bash
-ENVOY_DOCKER_BUILD_DIR=~/build ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev.server_only'
+ENVOY_DOCKER_BUILD_DIR=~/build ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev'
```
Comment (htuch) [inline ci/README.md]: This runs all the tests as well, is this intentional?

File: docs/install/sandboxes.rst [hunk 4]
```diff
@@ -297,10 +297,12 @@ This image has all software needed to build envoy. From your envoy directory::
 
   $ pwd
   src/envoy
-  $ docker run -t -i -v <SOURCE_DIR>:/source lyft/envoy-build:latest /bin/bash -c "cd /source && ci/do_ci.sh normal"
+  $ ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev'
```
Comment (htuch) [inline docs/install/sandboxes.rst]: Ditto.

Comment (ccaraman) [inline docs/install/sandboxes.rst]: It is intentional here.

File: ci/README.md (hunk 3 above)
Comment (ccaraman) [inline ci/README.md]: It is intentional as to match where the binary can be found.

File: docs/install/sandboxes.rst (hunk 4 above)
Comment (ccaraman) [inline docs/install/sandboxes.rst]: I will update the comment below to mention that it runs tests as well. And make the link to the readme to find other do_ci targets.

**Pull request merged**